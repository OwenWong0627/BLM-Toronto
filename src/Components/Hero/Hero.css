.hero-container {
   background-image: url("https://blacklivesmatter.ca/wp-content/uploads/2020/06/GettyImages-482191788-scaled.jpg");
   background-position: center top;
   background-size: cover;
   background-repeat: no-repeat;
   background-attachment: local;
   height: 87.5vh;
   width: 100%;
   display: flex;
   flex-direction: column;
   justify-content: center;
   align-items: center;
   box-shadow: inset 0 0 0 1000px rgba(50, 50, 50, 0.6);
   object-fit: contain;
 }
 
 .hero-container > h1 {
   color: #fff;
   font-size: 50px;
   text-align: center;
 }
 
 .hero-container > p {
   margin-top: 45px;
   color: rgba(255, 255, 255, 0.7);
   font-size: 32px;
   font-family: 'Trebuchet MS', 'Lucida Sans Unicode', 'Lucida Grande',
     'Lucida Sans', Arial, sans-serif;
   text-align: center;
   max-width: 75vw;
 }
 
 @media screen and (max-width: 960px) {
   .hero-container > h1 {
     font-size: 40px;
   }
   .hero-container > p {
      font-size: 26px;
    }
 }
 
 @media screen and (max-width: 768px) {
   .hero-container > h1 {
     font-size: 30px;
   }
 
   .hero-container > p {
     font-size: 20px;
   }
 }