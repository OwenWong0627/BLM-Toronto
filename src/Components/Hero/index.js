import React from 'react';
import './Hero.css';

function Hero() {
   return (
      <div className='hero-container'>
         <h1><font style={{ color: "#00BFFF" }}>DISCOVER</font> LOCAL BLACK OWNED BUSINESSES & BLACK ENTREPRENEURS</h1>
         <p>Lorem ipsum, dolor sit amet consectetur adipisicing elit. Dolore et minima, quis suscipit ullam iusto hic delectus ducimus qui minus.</p>
      </div>
   );
}

export default Hero;