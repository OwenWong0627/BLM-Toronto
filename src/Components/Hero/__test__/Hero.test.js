import React from 'react';
import ReactDOM from 'react-dom';
import Hero from '..';

import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';

afterEach(cleanup);
it('Renders without crashing', () => {
   const div = document.createElement("div");
   ReactDOM.render(<Hero />, div);
});

it('Renders the correct content', () => {
   const { getByTestId } = render(<Hero />);
   expect(getByTestId("hero")).toHaveTextContent("DISCOVER LOCAL BLACK-OWNED BUSINESSES & BLACK ENTREPRENEURS");
   expect(getByTestId("hero")).toHaveTextContent("Location Searching Tool to Locate Local Black-Owned Businesses");
});

it('matches snapshot', () => {
   const tree = renderer.create(<Hero />).toJSON();
   expect(tree).toMatchSnapshot();
});