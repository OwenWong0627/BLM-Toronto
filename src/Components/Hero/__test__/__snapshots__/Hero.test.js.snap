// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="hero-container"
  data-testid="hero"
>
  <h1>
    <font
      style={
        Object {
          "color": "#00BFFF",
        }
      }
    >
      DISCOVER
    </font>
     LOCAL BLACK-OWNED BUSINESSES & BLACK ENTREPRENEURS
  </h1>
  <p>
    Location Searching Tool to Locate Local Black-Owned Businesses
  </p>
</div>
`;
